### PR TITLE
{bio,devel}[foss/2019a,gompi/2019a] BLAST+ v2.9.0, Boost v1.67.0

### PR DESCRIPTION
**Pull request by smoors**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
nic427 - Linux centos linux 7.5.1804, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/f000bf8de781c898bdcc395597a1c53e for a full test report.

File: easybuild/easyconfigs/b/BLAST+/BLAST+-2.9.0-foss-2019a.eb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+source_urls = ['http://ftp.ncbi.nlm.nih.gov/blast/executables/%(namelower)s/%(version)s/']
+sources = ['ncbi-blast-%(version)s+-src.tar.gz']
+checksums = ['a390cc2d7a09422759fc178db84de9def822cbe485916bbb2ec0d215dacdc257']
+
+dependencies = [
+    ('zlib', '1.2.11'),
+    ('bzip2', '1.0.6'),
+    ('PCRE', '8.43'),
+    ('Boost', '1.67.0'),  # build fails with Boost-1.70.0
```
Comment (boegel) [inline easybuild/easyconfigs/b/BLAST+/BLAST+-2.9.0-foss-2019a.eb]: Introducing another `Boost` version in the `2019a` generation of easyconfigs is going to be a PITA...

How bad does the build fail? Is it fixable?

Comment (zao) [reply easybuild/easyconfigs/b/BLAST+/BLAST+-2.9.0-foss-2019a.eb]: All failures are related to Boost.Test use:

* `collector` changed name to `collector_t`,
* implementation detail `ignore_unused_variable_warning` moved to Boost.Core's `ignore_unused`,
* `log_build_info` grew a new argument with a default parameter.

A standalone build completes and passes `make check` with a patch that addresses these three problems.

**Comment by zao** [comment]
[BLAST+-2.9.0-boost-1.70-fixes.patch.txt](https://github.com/easybuilders/easybuild-easyconfigs/files/3217749/BLAST%2B-2.9.0-boost-1.70-fixes.patch.txt)

File: easybuild/easyconfigs/b/BLAST+/BLAST+-2.9.0-foss-2019a.eb (hunk 1 above)
Comment (smoors) [reply easybuild/easyconfigs/b/BLAST+/BLAST+-2.9.0-foss-2019a.eb]: challenge accepted :)

**Comment by smoors** [comment]
@zao thanks for the patch! I had not seen it, but it looks very similar to mine :)

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
nic18 - Linux centos linux 7.5.1804, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/59a4417c0e56d61058b937c9e7302d43 for a full test report.

File: easybuild/easyconfigs/b/BLAST+/BLAST+-2.9.0-foss-2019a.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/b/BLAST+/BLAST+-2.9.0-foss-2019a.eb]: :+1:

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in this PR)
node3119.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/55d91aa21bb412a3389cbdb3cc1761cf for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2401.golett.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/9fae215e74ca5030fff8e963d5a67948 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2010.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/7b0af09d6b4e93b209bca1d62ba83da9 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @smoors!

**Comment by boegel** [comment]
@smoors I think we can/should actually pull this down to `gompi/2019a`, since there's no BLAS/LAPACK involved here?

**Comment by boegel** [comment]
> @smoors I think we can/should actually pull this down to `gompi/2019a`, since there's no BLAS/LAPACK involved here?

Done in #8433

**Pull request merged**